File: solutions/lua/secret-handshake/2/secret-handshake.lua
return function(number)
  local result = setmetatable({}, { __index = table })
  if number & 16 > 0 then
    if number & 8 > 0 then result:insert('jump') end
    if number & 4 > 0 then result:insert('close your eyes') end
    if number & 2 > 0 then result:insert('double blink') end
    if number & 1 > 0 then result:insert('wink') end
  else
    if number & 1 > 0 then result:insert('wink') end
    if number & 2 > 0 then result:insert('double blink') end
    if number & 4 > 0 then result:insert('close your eyes') end
    if number & 8 > 0 then result:insert('jump') end
  end
  return result
end

File: solutions/ruby/secret-handshake/2/secret_handshake.rb
# frozen_string_literal: true

# class SecretHandshake
class SecretHandshake
  attr_reader :commands

  SHAKES = { 8 => 'jump', 4 => 'close your eyes', 2 => 'double blink', 1 => 'wink' }.freeze
  def initialize(number) =
    @commands = SHAKES.select { number.anybits?(_1) }
                      .values
                      .send(number.anybits?(16) ? :itself : :reverse)
end
